hypertrace-ingester/src/test/java/org/hypertrace/migration/MigrationTestRpc.java

```diff
@@ -0,0 +1,335 @@
+import io.jaegertracing.api_v2.JaegerSpanInternalModel.Span;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Stream;
+import org.hypertrace.core.datamodel.RawSpan;
+import org.hypertrace.core.semantic.convention.constants.error.OTelErrorSemanticConventions;
+import org.hypertrace.core.semantic.convention.constants.rpc.OTelRpcSemanticConventions;
+import org.hypertrace.core.span.constants.RawSpanConstants;
+import org.hypertrace.core.spannormalizer.jaeger.JaegerSpanNormalizer;
+import org.hypertrace.semantic.convention.utils.rpc.RpcSemanticConventionUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class MigrationTestRpc {
+
+  private final Random random = new Random();
+  private JaegerSpanNormalizer normalizer;
+
+  @BeforeEach
+  public void setup()
+      throws SecurityException, NoSuchFieldException, IllegalArgumentException,
+          IllegalAccessException {
+    // Since JaegerToRawSpanConverter is a singleton, we need to reset it for unit tests to
+    // recreate the instance.
+    Field instance = JaegerSpanNormalizer.class.getDeclaredField("INSTANCE");
+    instance.setAccessible(true);
+    instance.set(null, null);
+
+    // Create a JaegerSpanNormaliser
+    String tenantId = "tenant-" + this.random.nextLong();
+    Map<String, Object> configs = Map.of("processor", Map.of("defaultTenantId", tenantId));
+    this.normalizer = JaegerSpanNormalizer.get(ConfigFactory.parseMap(configs));
+  }
+
+  @Test
+  public void testGrpcFields() throws Exception {
+
+    Map<String, String> tagsMap =
+        new HashMap<>() {
+          {
+            put(RawSpanConstants.getValue(GRPC_ERROR_MESSAGE), "Some error message");
+            put(RawSpanConstants.getValue(CENSUS_RESPONSE_STATUS_CODE), "12");
+            put(RawSpanConstants.getValue(GRPC_STATUS_CODE), "13");
+            put(RawSpanConstants.getValue(CENSUS_RESPONSE_CENSUS_STATUS_CODE), "14");
+            put(
+                RawSpanConstants.getValue(CENSUS_RESPONSE_STATUS_MESSAGE),
+                "CENSUS_RESPONSE_STATUS_MESSAGE");
+            put(RawSpanConstants.getValue(ENVOY_GRPC_STATUS_MESSAGE), "ENVOY_GRPC_STATUS_MESSAGE");
+            put(RawSpanConstants.getValue(GRPC_REQUEST_BODY), "some grpc request body");
+            put(RawSpanConstants.getValue(GRPC_RESPONSE_BODY), "some grpc response body");
+          }
+        };
+
+    Span span = createSpanFromTags(tagsMap);
+    RawSpan rawSpan = normalizer.convert("tenant-key", span);
+
+    assertAll(
+        () ->
+            assertEquals(
+                rawSpan.getEvent().getGrpc().getResponse().getErrorMessage(),
+                RpcSemanticConventionUtils.getGrpcErrorMsg(rawSpan.getEvent())),
```

Comment:
In this method `getGrpcErrorMsg`, do we have to check for `contains key` before accessing value from map - `attributeValueMap.get(OTelErrorSemanticConventions.EXCEPTION_MESSAGE.getValue())` ? Or impl of map support already?

Reply:
In the method `getGrpcErrorMsg`, we are missing checking `RPC_ERROR_MESSAGE`.
As per stack order, OTEL. EXCEPTION_MESSAGE, RPC_ERROR_MESSAGE, and last is `GRPC_ERROR_MESSAGE` as per GrpcFieldsGenerator

Reply:
It checks first and then accesses 
`    if (isRpcTypeGrpcForOTelFormat(attributeValueMap)
        && attributeValueMap.get(OTelErrorSemanticConventions.EXCEPTION_MESSAGE.getValue())
            != null) {
      return attributeValueMap
          .get(OTelErrorSemanticConventions.EXCEPTION_MESSAGE.getValue())
          .getValue();
    }`

Reply:
Yes looks like `RPC_ERROR_MESSAGE` was missed as it was present in both `fieldGeneratorMap` and in `rpcFieldGeneratorMap`

Reply:
@sarthak77 lets add a dedicated test for testing `getGrpcErrorMsg`